src/ui/qt/qt_ui.h

```diff
@@ -12,6 +12,8 @@ class UiWindowQt : public UiWindow
 
     virtual void drawPoint( const Point2d & point, const PaintColor & color );
     virtual void drawLine( const Point2d & start, const Point2d & end, const PaintColor & color );
+    virtual void drawEllipse(const Point2d & center, const int xDiameter, const int yDiameter, const PaintColor & color);
```

Comment:
You make these functions as *virtual* but base class doesn't have them. We must add such functions in base class as well.